server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -108,19 +123,48 @@ public Task(
      *            generate data?
      */
     public final TaskInfo taskInfo(String localNodeId, boolean detailed) {
+        return taskInfo(localNodeId, detailed, detailed == false);
+    }
+
+    /**
+     * Build a version of the task status you can throw over the wire and back
+     * with the option to include resource stats or not.
+     * This method is only used during creating TaskResult to avoid storing resource information into the task index.
+     *
+     * @param excludeStats should information exclude resource stats.
+     *                     By default, detailed flag is used to control including resource information.
+     *                     But inorder to avoid storing resource stats into task index as strict mapping is enforced and breaks when adding this field.
+     *                     In the future, task-index-mapping.json can be modified to add resource stats.
+     */
+    private TaskInfo taskInfo(String localNodeId, boolean detailed, boolean excludeStats) {
         String description = null;
         Task.Status status = null;
+        TaskResourceStats resourceStats = null;
         if (detailed) {
             description = getDescription();
             status = getStatus();
         }
-        return taskInfo(localNodeId, description, status);
+        if (excludeStats == false) {
+            resourceStats = new TaskResourceStats(new HashMap<>() {
+                {
```

Comment:
We probably should not create a new class for that: `new TaskResourceStats(Collections.singletonMap(TOTAL, getTotalResourceStats()))`